### PR TITLE
Add and remove functions to the APP

### PR DESCRIPTION
**Pull request by AlierPM**
# In this project, the following changes have been effected:

* A user can add a new task
* A user can delete a task
* A user can edit a task
* Deleting a task updates all remaining items' indexes
* All changes to the To Do List are saved in local storage

[LIVE DEMO](https://alierpm.github.io/To-Do-List/dist/index.html)

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,57 +1,28 @@
...
-    <link
-      rel="stylesheet"
-      href="https://pro.fontawesome.com/releases/v5.15.1/css/all.css"
-    />
-    <link rel="stylesheet" href="style.css" type="text/css" />
-    <title>To-Do-List</title>
+    <meta charset="UTF-8">  
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<link href="style.css" rel="stylesheet">
+		<link rel="preconnect" href="https://fonts.googleapis.com">
```
Comment (MeqdamAlqudah) [inline src/index.html]: - [x] Please make sure that checking item as completed is saved in the localStorage.
image

File: src/index.html [hunk 2]
```diff
@@ -1,57 +1,28 @@
...
+		<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
Comment (MeqdamAlqudah) [inline src/index.html]: - [  ]  Please I can see that the user can edit a task but the issue is when the user edits two tasks a weird behavior happen one of the edited tasks seems to override editing the other one please fix this issue
image

File: src/index.html [hunk 3]
```diff
@@ -1,57 +1,28 @@
...
+		<link href="https://fonts.googleapis.com/css2?family=Lato:wght@100;300;400;700;900&display=swap" rel="stylesheet">
```
Comment (MeqdamAlqudah) [inline src/index.html]: - [ ] Please make sure to fix this linter error
image

**Review by MeqdamAlqudah — CHANGES_REQUESTED** [review]
Hi @AlierPM , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:+1: Readme File

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @AlierPM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions
- [ ] It would improve the user experience if the task caan be renamed by clicking on the text.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/removeTask.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import getLocalStorage from './getStorage.js';
+import setLocalStorage from './setStorage.js';
+import updateArray from './updateArray.js';
+
+const removeTask = (id) => {
+  let taskArray = getLocalStorage();
+  if (id >= 0) {
+    taskArray = taskArray.filter((task) => task.id !== id);
+  } else {
+    taskArray = taskArray.filter((task) => !task.completed);
```
Comment (rloterh) [inline src/modules/removeTask.js:10]: - [ ] Checking a task as completed does not change the corresponding task value of the local storage. Kindly ensure that checking a task as completed changes the value of that task from `false` to `true`, as informed in the previous review

File: src/modules/editTask.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import getLocalStorage from './getStorage.js';
+import setLocalStorage from './setStorage.js';
+
+const editTask = (id) => {
```
Comment (rloterh) [inline src/modules/editTask.js:4]: - [ ] Editing a task overrides the previous task, and task 2 rather gets edited in the local storage. Kindly ensure that the appropriate task edited reflects on the both the frontend and the in the corresponding local storage

**Before Editing First task `one`:**
image


**After Editing First take `one`:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Alier, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShahierNashaat — APPROVED** [review]
Hi @AlierPM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

# Approved :ok_hand: :ok_hand: :ok_hand:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**